### PR TITLE
Add compiler test and opts argument

### PR DESCRIPTION
**Pull request by chq-matteo**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
 import test from 'ava';
 import arduino from './';
 
+const fixture = path.join.bind(path, __dirname, 'fixtures');
 test('arduino loads arduino latest', async t => {
-  const arduinoObj = arduino();
+  const arduinoObj = arduino({tag: 'load'});
   const err = await pify(arduinoObj.load)();
   t.is(err, undefined);
   const accessErr = await pify(fs.access)(arduinoObj.binary(), fs.constants.X_OK);
```
Comment (simonepri) [inline test.js]: Please use this here:
https://www.npmjs.com/package/executable

**Comment by simonepri** [comment]
Marging

**Comment by chq-matteo** [comment]
There is a problem with the linux binary

**Comment by simonepri** [comment]
Maibe the bin path is wrong?

**Comment by simonepri** [comment]
It was wrong:
https://github.com/arduino-cli/arduino-node/commit/fb4a6bdada502a2dc6e23b72cc82fb19e08aa7f2

**Comment by simonepri** [comment]
image
🎉

**Comment by simonepri** [comment]
Still failing on linux can you check it?
It seems that no bin folder is created on linux.
On windows works

**Comment by chq-matteo** [comment]
https://downloads.arduino.cc/arduino-latest-linux64.tar.xz
We don't call 'em latest in linux we call 'em nightly

**Pull request merged**